src/test/java/uk/gov/hmcts/reform/sandl/snlrules/rules/listings/SessionIsOverlistedTests.java

```diff
@@ -0,0 +1,349 @@
+package uk.gov.hmcts.reform.sandl.snlrules.rules.listings;
+
+import org.drools.core.base.RuleNameEqualsAgendaFilter;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kie.api.runtime.KieSession;
+import org.mockito.runners.MockitoJUnitRunner;
+import uk.gov.hmcts.reform.sandl.snlrules.model.HearingPart;
+import uk.gov.hmcts.reform.sandl.snlrules.model.ProblemTypes;
+import uk.gov.hmcts.reform.sandl.snlrules.model.Session;
+import uk.gov.hmcts.reform.sandl.snlrules.services.DroolsService;
+
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.sandl.snlrules.rules.RulesTestHelper.assertProblems;
+import static uk.gov.hmcts.reform.sandl.snlrules.rules.RulesTestHelper.getInsertedProblems;
+import static uk.gov.hmcts.reform.sandl.snlrules.rules.RulesTestHelper.setDateInRules;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SessionIsOverlistedTests {
+
+    private final static String rulesDefinition = "Listings";
+
+    private DroolsService droolsService;
+    private KieSession rules;
+
+    @Before
+    public void setup() {
+        droolsService = new DroolsService(rulesDefinition);
+        droolsService.init();
+
+        rules = droolsService.getRulesSession();
+    }
+
+    @Test
+    public void should_be_no_problem_when_session_is_not_overlisted() {
+        setDateInRules(rules, 2018, 04, 1);
+
+        addSomeDummySessionsAndHearings();
+
+        droolsService.clearFactModifications();
+        rules.fireAllRules(new RuleNameEqualsAgendaFilter("Session is overlisted and 1 day or less before start"));
+        rules.fireAllRules(new RuleNameEqualsAgendaFilter("Session is overlisted greater or equal 50 percent and 1 to 3 days before start"));
+        rules.fireAllRules(new RuleNameEqualsAgendaFilter("Session is overlisted greater or equal 100 percent and 3 to 7 days before start"));
+        rules.fireAllRules(new RuleNameEqualsAgendaFilter("Session is listed less or equal 50 percent 7 days before start"));
+
+        assertProblems(droolsService,0, 0, 0);
+    }
+
+    @Test
+    public void should_be_no_problem_when_session_is_overlisted_more_than_month_or_less_before_start() {
+        String sessionId = "96462d68-76e3-430a-84c4-23983c448dc2";
```

Comment:
These should be put as const and shouldn't be duplicated.

Reply:
done